common/src/sql/dbinit.sql

```diff
@@ -1990,6 +1990,262 @@ CREATE TABLE omicron.public.role_assignment (
+    )
+);
+
+/* port settings groups included from port settings objects */
+CREATE TABLE omicron.public.port_settings_groups (
+    port_settings_id UUID,
+    port_settings_group_id UUID,
+
+    PRIMARY KEY (port_settings_id, port_settings_group_id)
+);
+
+CREATE TABLE omicron.public.port_settings_group (
+    id UUID PRIMARY KEY,
+    /* port settings in this group */
+    port_settings_id UUID NOT NULL,
+    name STRING(63) NOT NULL,
+    description STRING(512) NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    time_deleted TIMESTAMPTZ
+);
+
+CREATE UNIQUE INDEX ON omicron.public.port_settings_group (
+    name
+) WHERE
+    time_deleted IS NULL;
+
+CREATE TABLE omicron.public.port_settings (
+    id UUID PRIMARY KEY,
+    name STRING(63) NOT NULL,
+    description STRING(512) NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    time_deleted TIMESTAMPTZ
+);
+
+CREATE UNIQUE INDEX ON omicron.public.port_settings (
+    name
+) WHERE
+    time_deleted IS NULL;
+
+CREATE TYPE omicron.public.port_geometry AS ENUM (
+    '1xQSFP28',
+    '2xQSFP28',
+    '4xSFP28'
+);
+
+CREATE TABLE omicron.public.port_settings_port_config (
+    port_settings_id UUID PRIMARY KEY,
+    geometry omicron.public.port_geometry
+);
+
+CREATE TABLE omicron.public.port_settings_link_config (
+    port_settings_id UUID,
+    lldp_service_config_id UUID NOT NULL,
+    link_name TEXT,
+    mtu INT4,
+
+    PRIMARY KEY (port_settings_id, link_name)
+);
+
+CREATE TABLE omicron.public.lldp_service_config (
+    id UUID PRIMARY KEY,
+    lldp_config_id UUID,
+    enabled BOOL NOT NULL
+);
+
+CREATE TABLE omicron.public.lldp_config (
+    id UUID PRIMARY KEY,
+    name STRING(63) NOT NULL,
+    description STRING(512) NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    time_deleted TIMESTAMPTZ,
+    chassis_id TEXT,
+    system_name TEXT,
+    system_description TEXT,
+    management_ip TEXT
+);
+
+CREATE UNIQUE INDEX ON omicron.public.lldp_config (
+    name
+) WHERE
+    time_deleted IS NULL;
+
+CREATE TYPE omicron.public.interface_kind AS ENUM (
+    'primary',
+    'vlan',
+    'loopback'
+);
+
+CREATE TABLE omicron.public.port_settings_interface_config (
+    port_settings_id UUID,
+    id UUID PRIMARY KEY,
+    interface_name TEXT NOT NULL,
+    v6_enable BOOL NOT NULL,
+    kind omicron.public.interface_kind,
+
+    CONSTRAINT port_settings_interface_name_unique UNIQUE (
+        port_settings_id, interface_name
+    )
+);
+
+CREATE UNIQUE INDEX ON omicron.public.port_settings_interface_config (
+    port_settings_id, interface_name
+);
+
+CREATE TABLE omicron.public.vlan_interface_config (
+    interface_config_id UUID,
+    vid INT4,
+
+    PRIMARY KEY (interface_config_id, vid)
+);
+
+CREATE TABLE omicron.public.port_settings_route_config (
+    port_settings_id UUID,
+    interface_name TEXT,
+    dst INET,
+    gw INET,
```

Comment:
If this is `NULL` does that mean the gateway is on-link? I understand it's not currently nullable given that it's part of the primary key, but I'm not sure we want to do that.

Reply:
I don't think on-link routes will go in this table. We may have interface routes that automatically resolve to an IPv6 link local gateway address and then put that address in this table, but my current thinking is there will always be an address for `gw`.